apps/web/src/views/Mev/hooks/index.ts

```diff
@@ -1,59 +1,62 @@
 import { ChainId } from '@pancakeswap/chains'
 import { useQuery } from '@tanstack/react-query'
 import useActiveWeb3React from 'hooks/useActiveWeb3React'
-import { useCallback, useMemo } from 'react'
+import { useCallback } from 'react'
+import { WalletClient } from 'viem'
 import { bsc } from 'viem/chains'
 
-async function fetchMEVStatus(): Promise<boolean> {
-  if (!window.ethereum || (!window.ethereum as any)?.request) {
-    throw new Error('Ethereum provider not found')
+import { useWalletClient } from 'wagmi'
+
+async function fetchMEVStatus(walletClient: WalletClient): Promise<{ mevEnabled: boolean; isError: boolean }> {
+  if (!walletClient || !walletClient.request) {
+    console.error('Ethereum provider not found')
+    return { mevEnabled: false, isError: true }
   }
 
   try {
-    const result = await (window.ethereum as any)?.request({
+    const result = await walletClient.request({
+      // @ts-ignore
       method: 'eth_call',
       params: [
         {
+          from: walletClient.account?.address ?? '0x',
           to: '0x0000000000000000000000000000000000000048',
           value: '0x30',
+          data: '0x',
         },
       ],
     })
-    return result === '0x30'
+    return { mevEnabled: result === '0x30', isError: false }
   } catch (error) {
-    console.error('Error checking MEV status:', error)
-    return false
+    if ((error as any).code === -32601) console.error('wallet_addEthereumChain is not supported')
```

Comment:
I don't quite understand the logic here. Isn't this request checking wether custom rpc endpoint is set up? Don't see why it has anything to do with `wallet_addEthereumChain`

Reply:
yeah, it seems the eth_call is enough to check, let me move this to when user actually click the add wallet button